src/uct/ib/base/ib_device.c

```diff
@@ -913,50 +913,44 @@ uct_ib_device_roce_gid_to_sockaddr(sa_family_t af, const void *gid,
 }
 
 static ucs_status_t
-uct_ib_device_parse_subnet_list(const ucs_config_allow_list_t *subnet_strs,
-                                uct_ib_device_subnet_array_t *subnets)
+uct_ib_device_parse_subnet_filter(const ucs_config_allow_list_t *subnet_strs,
+                                  uct_ib_device_subnet_array_t *subnets)
 {
-    static const unsigned subnet_num_elems = 2;
-    static const unsigned sscanf_res_ok    = 1;
-    char *address_str, *mask_str, *subnet_str_dup;
+    static const size_t max_subnet_str_len = 60;
+    char *address_str, *mask_str;
     char **subnet_str;
+    char subnet_str_dup[max_subnet_str_len];
```

Comment:
`char subnet_str[UCS_SOCKADDR_STRING_LEN];`
no need to define max_subnet_str_len